src/ShellCheck/Analytics.hs

```diff
@@ -3096,5 +3097,19 @@ checkRedirectionToCommand _ t =
                 warn id 2238 "Redirecting to/from command name instead of file. Did you want pipes/xargs (or quote to ignore)?"
         _ -> return ()
 
+prop_checkNullaryExpansionTest1 = verify checkNullaryExpansionTest "[[ $(a) ]]"
+prop_checkNullaryExpansionTest2 = verify checkNullaryExpansionTest "[[ $a ]]"
+prop_checkNullaryExpansionTest3 = verifyNot checkNullaryExpansionTest "[[ $a=1 ]]"
+prop_checkNullaryExpansionTest4 = verifyNot checkNullaryExpansionTest "[[ -n $(a) ]]"
+checkNullaryExpansionTest _ t =
+    case t of
+        TC_Nullary _ _ (T_NormalWord _ (t2:[])) ->
```

Comment:
Might as well generalize this to any `T_NormalWord` to catch things like `[[ $a$b ]]`.

It could be gated on `not . isConstant`  instead of `isQuoteableExpansion` in that case.

Reply:
Followed your suggestion :)